### PR TITLE
Fix docstrings for time series forecasting.

### PR DESCRIPTION
**Pull request by nick863**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/tabular/forecasting_job.py [hunk 1]
```diff
@@ -78,10 +111,22 @@ def primary_metric(self, value: Union[str, ForecastingPrimaryMetrics]):
 
     @property
     def training(self) -> ForecastingTrainingSettings:
+        """
+        Return the forecast training settings.
+
+        :return: training settings.
+        :rtype: ForecastingTrainingSettings
```
Comment (diondrapeck) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/tabular/forecasting_job.py:118]:
```suggestion
        :rtype: ~azure.ai.ml.automl.ForecastingTrainingSettings
```

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/tabular/forecasting_job.py [hunk 2]
```diff
@@ -62,11 +83,23 @@ def __init__(
...
         return self._primary_metric
 
     @primary_metric.setter
-    def primary_metric(self, value: Union[str, ForecastingPrimaryMetrics]):
+    def primary_metric(self, value: Union[str, ForecastingPrimaryMetrics]) -> None:
+        """
+        Set the primary metric to use for model selection.
+
+        :param value: The primary metric for model selection.
+        :type: Union[str, ForecastingPrimaryMetrics]
```
Comment (diondrapeck) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/tabular/forecasting_job.py:101]:
```suggestion
        :type: Union[str, ~azure.ai.ml.automl.ForecastingPrimaryMetrics]
```

**Review by diondrapeck — CHANGES_REQUESTED** [review]

**Review by diondrapeck — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_restclient/v2023_04_01_preview/models/_azure_machine_learning_workspaces_enums.py [hunk 3]
```diff
@@ -627,8 +627,8 @@ class ForecastingModels(str, Enum, metaclass=CaseInsensitiveEnumMeta):
     #: This method is suitable for forecasting when data is stationary/non stationary, and
     #: multivariate with any type of data pattern, i.e., level/trend /seasonality/cyclicity.
     ARIMAX = "Arimax"
-    #: TCNForecaster: Temporal Convolutional Networks Forecaster. //TODO: Ask forecasting team for
-    #: brief intro.
+    #: TCNForecaster: Temporal Convolutional Networks Forecaster hierarchically captures relationships
+    #: at low-, intermediate- and high- time scales to produce the forecast.
```
Comment (EricWrightAtWork) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_restclient/v2023_04_01_preview/models/_azure_machine_learning_workspaces_enums.py:631]: Would change this to something like: Deep neural network model capable of modeling correlations over long time periods.

Comment (nick863) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_restclient/v2023_04_01_preview/models/_azure_machine_learning_workspaces_enums.py]: Fixed.

**Review by EricWrightAtWork — APPROVED** [review]

**Pull request merged**